fastlane/fastlane
# [action][spm] deprecate `build_path` option in favor of `scratch_path`, as recommended by Swift CLI

### Checklist

### Motivation and Context

The swift cli has been warning about the build-path deprecation.

### Description

This PR adds the option to use scratch_path instead of build_path, but it kept the build_path available since the swift cli is still accepting it as a warning.
I tested it, just changing the option.

### Testing Steps
Using this PR as the Fastlane source.

1. Create an SPM project 
2. Initialise fastlane 
3. Add to their Fastfile the spm action with scratch_path instead of build_path
4. Change scratch_path to build_path and it should work as well.
